docs/code-style.md

```diff
@@ -0,0 +1,667 @@
+* [Custom _HLint_ rules](https://github.com/input-output-hk/cardano-sl/blob/master/HLint.hs)
+
+General guide lines
+-------------------
+### Line Length
+
+Maximum line length is *80 characters* or *100 characters* if necessary.
+
+Modern screens have high definition and big width.
+But with some tiling managers and two terminals on one screen you are not able to
+see many characters on one line.
+On the other hand, restricting line size to a very small number like 80 leads to
+some crazy indentation despite the fact that
+shorter lines should force you to write well structured code.
+That's why *100* is a reasonable compromize.
+
+
+### Indentation
+
+Tabs are illegal.  Use spaces for indenting.  Indent your code blocks
+with *4 spaces*.  Indent the `where` keyword with two spaces to set it
+apart from the rest of the code and indent the definitions in a
+`where` clause with 2 spaces.  Some examples:
+
+```haskell
+sayHello :: IO ()
+sayHello = do
+    name <- getLine
+    putStrLn $ greeting name
+  where
+    greeting name = "Hello, " ++ name ++ "!"
+
+filter :: (a -> Bool) -> [a] -> [a]
+filter _ []     = []
+filter p (x:xs)
+    | p x       = x : filter p xs
+    | otherwise = filter p xs
+```
+
+### Blank Lines
+
+One blank line between top-level definitions. No blank lines between
+type signatures and function definitions. Add one blank line between
+functions in a type class instance declaration if the function bodies
+are large. You can add blank lines inside a big `do` block to separate logical
+parts of it. You can also use blank lines to separate definitions inside `where`
+clause. Blank lines are used to separate imports (this is described in section below).
+
+### Whitespace
+
+Surround binary operators with a single space on either side. In case of
+currying add one space between the argument and the operation.
+
+_Exceptions (where you're allowed to not follow this rule)_:
+
+1. `%` operator for string formatting from `Formatting` library. Compare:
+
+`("block with "%int%" entries") cnt`
+
+`("block with " % int % " entries") cnt`
+
+2. `+|`-like operators from [`fmt`](http://hackage.haskell.org/package/fmt) library. Compare
+
+`"block with "+|cnt|+" entries"`
+
+`"block with " +| cnt |+ " entries"`
+
+So, you can omit spaces if surrounding strings have spaces. It also helps in
+cases when formatted strings are too long.
+
+Use some tool to remove trailing spaces!
+
+### Naming convention
+
+Casing:
++ **_lowerCamelCase_** for function and variable names.
++ **_UpperCamelCase_** for types.
++ TODO: some convention for global constants?
+
+Don't use short names like `n`, `sk`, `f` unless their meaning is clear from
+context (function name, types, other variables, etc.).
+
+For readability reasons, don't capitalize all letters when using an
+abbreviation.  For example, write `HttpServer` instead of
+`HTTPServer`.  Exception: two or three letter abbreviations, e.g. `IO`, `STM`, or established 
+names like `SQLite`.
```

Comment:
It's good you included `SQLite` here, it made me doubt a lot when we first used it.